src/bin/common/solving_loop.ml

```diff
@@ -837,21 +857,14 @@ let main () =
 
       | {contents = `Get_model; _ } ->
         if Options.get_interpretation () then
-          match State.get partial_model_key st with
-          | Some Model ((module SAT), partial_model) ->
-            begin
-              match SAT.get_model partial_model with
-              | Some (lazy model) ->
-                Models.output_concrete_model
-                  (Options.Output.get_fmt_regular ()) model;
-                st
-              | _ ->
-                (* TODO: is it reachable? *)
-                st
-            end
-          | None ->
-            (* TODO: add the location of the statement. *)
-            recoverable_error "No model produced."; st
+          let _ = match State.get partial_model_key st with
+            | Some model ->
+              print_model (Options.Output.get_fmt_regular ()) model
+            | None ->
+              (* TODO: add the location of the statement. *)
+              recoverable_error "No model produced."
+          in
+          st
```

Comment:
Not convinced we should print the unknown reason when we have a model, it will just be noise. If the user wants to see the unkonwn reason, they can call `(get-info :unknown-reason)`, we shouldn't force it down their throat.

(For `--dump-models` I am not entirely convinced either but it kind of makes sense b/c the user has no control & it helps distinguish the intermediate models from the final models with `--interpretation every`, but for `(get-model)` there is a mechanism in place to display that information with `(get-info :unknown-reason)` already)